src/js/components/Select/index.d.ts

```diff
@@ -26,17 +39,17 @@ export interface SelectProps {
   dropTarget?: object;
   dropProps?: DropProps;
   focusIndicator?: boolean;
-  icon?: boolean | ((...args: any[]) => any) | React.ReactNode;
+  icon?: boolean | (() => React.ReactElement) | React.ReactNode;
   id?: string;
-  labelKey?: string | ((...args: any[]) => any);
+  labelKey?: string | ((option: { label: string; value: number }) => object);
   margin?: MarginType;
   messages?: { multiple?: string };
   multiple?: boolean;
   name?: string;
-  onChange?: (...args: any[]) => void;
-  onClose?: (...args: any[]) => any;
-  onMore?: (...args: any[]) => any;
-  onOpen?: (...args: any[]) => any;
+  onChange?: (event: onChangeEvent) => void;
+  onClose?: () => void;
+  onMore?: () => void;
+  onOpen?: () => void;
```

Comment:
This change needs to be tested using those props, I think we need to keep the args.